lib/iris/_lazy_data.py

```diff
@@ -419,3 +419,21 @@ def map_complete_blocks(src, func, dims, out_sizes):
         out_chunks[dim] = size
 
     return data.map_blocks(func, chunks=out_chunks, dtype=src.dtype)
+
+
+def sample(array):
+    """
+    Realise and return the smallest possible section of a lazy array.
+
+    Intended for checking masks and fill values - dask.array.ma arrays have no
+    distinguishing properties from normal dask arrays, so must unfortunately
```

Comment:
Note that dask arrays seem to keep track of the array type of their chunks, I think there may be no need to sample. This code will give you the array type:
```python
type(dask.array.utils.meta_from_array(array))
```